generators/app/index.js

```diff
@@ -27,65 +33,76 @@ module.exports = Generator.extend({
         name: 'githubUsername',
         message: 'What is the github project organisation or username?',
         validate: required,
-        default: githubUsername
+        default: githubUsername,
+        when: utils.noConfig('githubUsername', this.initialConfig)
```

Comment:
I refact the code to use `isEmpty(this.initialConfig['githubUsername'])`

Also added some unit tests.